### PR TITLE
Enable StyleCop rules

### PR DESCRIPTION
**Pull request by pgrawehr**
Finally enables the rules, as documented in #816

### CODE REVIEW

**Comment by pgrawehr** [comment]
After this is merged, the build for all the currently open PRs should be restarted (and then checked whether they need a rebase)

File: eng/Analyzers.props [hunk 1]
```diff
@@ -3,10 +3,10 @@
 <!-- If the runtime is unknown, GeneratePlatformNotSupportedAssembly is set in 
 System.Device.Gpio.csproj, which creates an auto-generated code 
 file which is not marked as such, making StyleCop fail. -->
-  <PropertyGroup Condition="'$(MSBuildProjectExtension)' == '.csproj' AND '$(GeneratePlatformNotSupportedAssembly)'!='true' AND '$(MSBuildProjectFile)'!='Iot.Device.Bindings.csproj'">
+  <PropertyGroup Condition="'$(MSBuildProjectExtension)' == '.csproj' AND '$(GeneratePlatformNotSupportedAssembly)'!='true'">
```
Comment (krwq) [inline eng/Analyzers.props]:
```suggestion
  <PropertyGroup Condition="'$(MSBuildProjectExtension)' == '.csproj' AND '$(GeneratePlatformNotSupportedAssembly)' != 'true'">
```

File: eng/Analyzers.props [hunk 2]
```diff
@@ -3,10 +3,10 @@
...
     <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)\StyleCop.Analyzers.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
-  <ItemGroup Condition="'$(MSBuildProjectExtension)' == '.csproj' AND '$(GeneratePlatformNotSupportedAssembly)'!='true' AND '$(MSBuildProjectFile)'!='Iot.Device.Bindings.csproj'">
+  <ItemGroup Condition="'$(MSBuildProjectExtension)' == '.csproj' AND '$(GeneratePlatformNotSupportedAssembly)'!='true'">
```
Comment (krwq) [inline eng/Analyzers.props]:
```suggestion
  <ItemGroup Condition="'$(MSBuildProjectExtension)' == '.csproj' AND '$(GeneratePlatformNotSupportedAssembly)' != 'true'">
```

**Comment by krwq** [comment]
Thank you @pgrawehr! This is gonna save me a bunch of time on style comments :smile:

**Comment by pgrawehr** [comment]
@krwq: I hope so. And it will save us contributors hopefully some silly back-and-forth for only that. 

Should we write the most important style rules into the Contributing file (to avoid people getting frustrated over some messages) or do you expect that everyone knows how to check what a rule means (ie. how to use google?)

**Comment by krwq** [comment]
@pgrawehr https://github.com/dotnet/iot/blob/master/Documentation/Coding-guidelines.md

**Comment by pgrawehr** [comment]
Some messages may not always be very intuitive. For instance it says "Braces should not be omitted" if you forget to use {} after if/for/etc (instead of what you should do: "Add braces after...") Or it complains about spaces at the end of empty lines if the line contains nothing but spaces.

**Comment by krwq** [comment]
@pgrawehr I think we can help on that if someone asks, I doubt people will read whole guidelines, likely will paste the error in their favorite search engine...

**Pull request merged**